db/repair_bucket.go

```diff
@@ -75,82 +76,170 @@ func (r *RepairBucket) InitFrom(params RepairBucketParams) *RepairBucket {
 	return r
 }
 
+/*
+
+This is how the view is iterated:
+
+┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+                ┌ ─ ─ ─│─ ─ ─ ─ ─ ─ ─ ─
+│                               ┌ ─ ─ ─│─ ─ ─ ─ ─ ─ ─ ┐
+                │      │                ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┐
+│┌────┐  ┌────┐  ┌────┐  ┌────┐ │┌────┐│ ┌────┐       │
+ │doc1│  │doc2│ ││doc3││ │doc4│  │doc5│ ││doc6│               │
+│└────┘  └────┘  └────┘  └────┘ │└────┘│ └────┘       │
+                │      │                │                     │
+└ ─ ─ ─ ─ ─ ▲ ─ ─ ─ ─ ─         │      │              │
+            │   └ ─ ─ ─ ─ ─ ▲ ─ ─ ─ ─ ─ │                     │
+            │               │   └ ─ ─ ─ ─ ─▲─ ─ ─ ─ ─ ┘
+      StartKey: ""          │           └ ─│─ ─ ─ ─ ─ ─ ─ ─ ─ ┘
+        Limit: 3            │              │       ▲
+    NumProcessed: 3         │              │       │
+                    StartKey: "doc3"       │       │
+                        Limit: 3           │       │
+                    NumProcessed: 2        │       └────────┐
+                                           │                │
+                                   StartKey: "doc5"         │
+                                       Limit: 3             │
+                                   NumProcessed: 1          │
+                                                            │
+                                                    StartKey: "doc6"
+                                                        Limit: 3
+                                                    NumProcessed: 0
+
+* It starts with an empty start key
+* For the next page, it uses the last key processed as the new start key
+* Since the start key is inclusive, it will see the start key twice (on first page, and on next page)
```

Comment:
Is it worth utilising startkey_docid here? https://blog.couchbase.com/pagination-couchbase/

Reply:
Checking that out

Reply:
@JFlath `startkey_docid` doesn't look relevant here.  It mainly looks relevant when indexing on a different key than the doc id, but in this case it's indexing the docid itself*, so there is no difference between `startkey` and `startkey_docid` 

*actually, it's indexing based on a compound key of `[sync_meta_exists, docid]`, but the same argument still holds

Reply:
Ahh yes, I was misremembering what the view emitted, my bad. While we're on the subject of pagination though, would adding a `skip=1` be any better than discarding the first result in SG? I suppose it would mean special-casing the first query though...

Reply:
> would adding a skip=1 be any better than discarding the first result in SG? I suppose it would mean special-casing the first query though...

That might make things a bit cleaner, even with the special casing.  Thanks for the suggestion!